src/useWindowSize.ts

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from 'react';
+
+interface WindowSize {
+  width: number;
+  height: number;
+}
+
+/**
+ * useWindowSize : 브라우저 창의 너비와 높이를 제공하는 훅
+ * @param {boolean} [isDelay=true] 딜레이 여부 판단. Default=true
+ * @param {number} [delayTime=200] 지연 시간(밀리초). isDelay가 true일 때 사용됨.Default=200
+ * @returns {{ width: number, height: number }} 창의 너비와 높이를 담은 객체
+ * @description
+ * 브라우저 창의 사이즈가 변경될 때마다 해당 창의 너비와 높이를 업데이트합니다.
+ * delay유무와 delayTime을 설정할 수 있습니다.
+ * UI의 세밀한 조작 등 동적인 변화에 유용합니다.
+ */
+
+const useWindowSize = (
+  isDelay: boolean = true,
+  delayTime: number = 200
+): WindowSize => {
+  const [windowSize, setWindowSize] = useState<WindowSize>({
+    width: window.innerWidth,
+    height: window.innerHeight,
+  });
+
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout | undefined;
+    const handleWindowResize = () => {
+      if (timeoutId) clearTimeout(timeoutId);
+
+      const updateWindowSize = () => {
+        setWindowSize({
+          width: window.innerWidth,
+          height: window.innerHeight,
+        });
+      };
+
+      if (isDelay) {
```

Comment:
```suggestion
      if (!!delayTime) {
```

[P2] `isDelay` 여부는 delayTime으로 체크해 볼 수 있을까요? 그럼 파라미터 수도 줄일 수 있을 것 같습니다!

Reply:
만약 알려주신 방식대로 `delayTime`만 넘기는 단일 파라미터 형식으로 변경한 후

`!!delayTime`을 사용하면 `Redundant double negation.eslint[no-extra-boolean-cast](https://eslint.org/docs/latest/rules/no-extra-boolean-cast)`
위 eslint 오류가 발생해서

delayTime이 양수일때만 실행하도록 하는 방법으로 수정해봤습니다

1. 상단에서 음수일 경우 에러로 넘기고
```ts
 if (delayTime < 0) {
    throw new Error("delayTime은 0 혹은 양수여야 합니다.");
  }
```

2. 0이상일 경우 판별
```ts
  if (delayTime > 0) {
    timeoutId = setTimeout(updateWindowSize, delayTime);
  } else {
    updateWindowSize();
  }
```
- 아무것도 넘기지 않았을 때 200의 delayTime으로 setTimeout설정
- 0을 넘기면 setTimeout이 설정되지 않음
- 0보다 큰 수를 넘기면 해당 delayTime으로 setTimeout 설정

결과적으로는 덕분에 파라미터 수를 하나 줄일 수 있게 되었네요😊

----

또 별개로 딜레이에 대해서는 고민이 되었던 점이 있습니다

> Default값으로 딜레이를 주는게, 또 얼마나 딜레이를 주는게 가장 적절한 방식인가?

이부분에 대해서는 다들 어떻게 생각하시는지 궁금합니다

Reply:
사람이 인지할 수 있는 최소 시간 단위에 맞춰 기본값으로 100정도가 좋다고 생각됩니다

Reply:
> 또 별개로 딜레이에 대해서는 고민이 되었던 점이 있습니다
> 
> > Default값으로 딜레이를 주는게 적절한가? 준다면 얼마나 딜레이를 주는게 가장 적절한 방식인가?
> 
> 이부분에 대해서는 다들 어떻게 생각하시는지 궁금합니다

사람이 인지할 수 있는 최소 시간 단위에 맞춰 기본값으로 100~200 정도가 좋다고 생각됩니다.

Reply:
@foresec 넵 저도 Default 값은 200정도가 적당하다고 생각합니다!